### PR TITLE
Fix a bug that sourcepos of custom_block is wrong

### PR DESCRIPTION
**Pull request by ohakutsu**
Closes https://github.com/increments/qiita_marker/issues/29

## What

- Fix a bug that sourcepos of custom_block is wrong.
    - https://github.com/increments/qiita_marker/issues/29
- Refactor `xml_attr_buff` to not hold in `qfm_custom_block` node.
    - Changed to a better implementation method. (use `cmark_strbuf_detach`)
- Remove `.rubocop_todo.yml`.
    - Temporarily added by https://github.com/increments/qiita_marker/pull/5
    - It is no longer needed due to merge of upstream.

### CODE REVIEW

File: ext/qiita_marker/qfm_custom_block.c [hunk 1]
```diff
@@ -114,29 +78,42 @@ static cmark_node *try_opening_qfm_custom_block_block(
   if (!indented && parent_type != CMARK_NODE_QFM_CUSTOM_BLOCK) {
     bufsize_t matched = scan_open_qfm_custom_block_fence(
         input, len, cmark_parser_get_first_nonspace(parser));
-    if (!matched) {
```
Comment (atm-snag2) [inline ext/qiita_marker/qfm_custom_block.c:117]: この早期リターンはそのままに近い形でもいいのではないか？

**Comment by kyntk** [comment]
I thought it would be helpful to describe the cause of the difference in the sourcepos as well.

File: ext/qiita_marker/qfm_custom_block.c [hunk 2]
```diff
@@ -81,26 +53,19 @@ static void free_node_qfm_custom_block(cmark_mem *mem, void *ptr) {
...
-  cmark_strbuf_free(cb->xml_attr_buff);
   mem->free(cb);
 }
 
 static int matches(cmark_syntax_extension *self, cmark_parser *parser,
                    unsigned char *input, int len,
                    cmark_node *parent_container) {
-  int res = 0;
-
-  if (get_qfm_custom_block_opening(parent_container)) {
```
Comment (kyntk) [inline ext/qiita_marker/qfm_custom_block.c:93]: Is it possible to contain a custom block as a child when another custom block is the parent?

Comment (ohakutsu) [reply ext/qiita_marker/qfm_custom_block.c:93]: Now, a custom block can contain another custom block, but this is not a specification and may change in the future.

File: ext/qiita_marker/qfm_custom_block.c [hunk 3]
```diff
@@ -120,21 +87,33 @@ static cmark_node *try_opening_qfm_custom_block_block(
...
+    cmark_strbuf_trim(info);
     set_qfm_custom_block_info(custom_block_node, (char *)info->ptr);
 
     cmark_node_set_syntax_extension(custom_block_node, self);
     cmark_parser_advance_offset(parser, (char *)input,
-                                len - cmark_parser_get_offset(parser), 0);
+                                cmark_parser_get_first_nonspace(parser) +
+                                    matched + info_len -
+                                    cmark_parser_get_offset(parser),
+                                0);
```
Comment (ohakutsu) [inline ext/qiita_marker/qfm_custom_block.c:116]: [memo]
`offset` was advanced to include the length of the newline character, which is why the `sourcepos` was incorrect.

**Review by kyntk — APPROVED** [review]
LGTM!

**Review by atm-snag2 — APPROVED** [review]
LGTM

**Pull request merged**